src/main/java/uk/gov/ons/census/casesvc/utility/QidCreator.java

```diff
@@ -53,35 +53,39 @@ private int calculateQuestionnaireType(String treatmentCode) {
       log.with("treatment_code", treatmentCode).error(UNKNOWN_COUNTRY_ERROR);
       throw new IllegalArgumentException();
     }
-    int questionnaireType = -1;
+
     if (treatmentCode.startsWith("HH")) {
-      if (country.equals("E")) {
-        questionnaireType = 1;
-      } else if (country.equals("W")) {
-        questionnaireType = 2;
-      } else if (country.equals("N")) {
-        questionnaireType = 4;
+      switch (country) {
+        case "E":
+          return 1;
+        case "W":
+          return 2;
+        case "N":
+          return 4;
```

Comment:
Is it worth adding a default here in case the country doesn't get matched to log and error with the offending country character, rather than just hitting the runtime exception below?

Reply:
Look at the top of the function - it checks the country then throws a meaningful error